lib/drivers/medtronic600Driver.js

```diff
@@ -0,0 +1,1221 @@
+const debug = isBrowser ? require('../bows')('Medtronic600Driver') : console.log;
+
+class ExtendableError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = this.constructor.name;
+    this.message = message;
+    if (typeof Error.captureStackTrace === 'function') {
+      Error.captureStackTrace(this, this.constructor);
+    } else {
+      this.stack = (new Error(message)).stack;
+    }
+  }
+}
+
+class TimeoutError extends ExtendableError {}
+
+class BCNLMessage {
+  constructor(bytes) {
+    if (new.target === BCNLMessage) {
+      throw new TypeError(`Cannot construct ${new.target.name} instances directly`);
+    }
+
+    this.payload = Buffer.from(bytes);
+  }
+
+  static get USB_BLOCKSIZE() {
+    return 64;
+  }
+  static get MAGIC_HEADER() {
+    return 'ABC';
+  }
+  static get READ_TIMEOUT_MS() {
+    return 4000;
+  }
+  static get SEND_DELAY_MS() {
+    return 500;
+  }
+
+  static get ASCII_CONTROL() {
+    return {
+      ACK: 0x06,
+      CR: 0x0D,
+      ENQ: 0x05,
+      EOT: 0x04,
+      ETB: 0x17,
+      ETX: 0x03,
+      LF: 0x0A,
+      NAK: 0x15,
+      STX: 0x02,
+    };
+  }
+
+  static wait(delay) {
+    // eslint-disable-next-line no-unused-vars
+    return new Promise((resolve, reject) => {
+      setTimeout(resolve, delay);
+    });
+  }
+
+  toString() {
+    return this.payload.toString('hex');
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  readMessage(hidDevice, readTimeout = BCNLMessage.READ_TIMEOUT_MS) {
+    return new Promise((resolve, reject) => {
+      const abortTimer = setTimeout(() => {
+        debug('TIMEOUT');
+        reject(new TimeoutError('Timeout error.'));
+      }, readTimeout);
+
+      let message = Buffer.alloc(0);
+
+      async.doWhilst(
+        (callback) => {
+          hidDevice.receive((raw) => {
+            const packet = Buffer.from(new Uint8Array(raw));
+
+            // Only process if we get data
+            if (packet.length === 0) {
+              return callback(false);
+            }
+
+            const header = packet.slice(0, 3).toString('ascii');
+            const size = packet[3];
+
+            if (header !== BCNLMessage.MAGIC_HEADER) {
+              debug('Invalid packet from Contour device');
+              clearTimeout(abortTimer);
+              return callback(new Error('Invalid USB packet received.'));
+            }
+
+            message = Buffer.concat([message, packet.slice(4)],
+              (message.length + packet.length) - 4);
+
+            // USB_BLOCKSIZE - 4, because we don't include the MAGIC_HEADER or the size byte
+            if (size < (BCNLMessage.USB_BLOCKSIZE - 4)) {
+              clearTimeout(abortTimer);
+              return callback(true);
+            }
+            return callback(false);
+          });
+        },
+        (valid) => {
+          if (valid instanceof Error) {
+            throw valid;
+          }
+          return (valid !== true);
```

Comment:
I was just being really explicit (using the type check as well), but since the `instanceof Error` catches the only other case, I think `!valid` is fine. @jebeck, is there a preference for this?